### PR TITLE
Registry api code migration

### PR DESCRIPTION
**Pull request by alexdunnjpl**
## 🗒️ Summary
Migrates code previously resident in [registry-api's support subtree](https://github.com/NASA-PDS/registry-api/tree/main/support/provenance)

## ⚙️ Test Data and/or Report
No tests (yet).

Dockerfile has been modified for faster/easier dev experience.

## ♻️ Related Issues
fixes #1 
#2 partially outstanding

### CODE REVIEW

File: src/pds/provenance/provenance.py [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+        username: str,
+        password: str,
+        verify_host_certs: bool = False,
+        log_filepath: Union[str, None] = None,
+        log_level: int = logging.INFO):
+    configure_logging(filepath=log_filepath, log_level=log_level)
+
+    log.info('starting CLI processing')
+
+    host = HOST(cluster_nodes, password, base_url, username, verify_host_certs)
```
Comment (nutjob4life) [inline src/pds/provenance/provenance.py:104]: Oh so that's how you use named tuple 😄

**Review by nutjob4life — APPROVED** [review]
LGTM 👍

File: docker/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+# ECS/Fargate. Refer to the script for its run-time assumptions under
+# which is it executed in the AWS environment.
+#
+# Basis
+# -----
+#
+# We use a slim python-enabled base image.
+FROM python:3.10-slim-bullseye
+
+ARG GIT_BRANCH=main
```
[the remark this answers is not in the document]
Comment (alexdunnjpl) [reply docker/Dockerfile:45]: Good catch, tyvm

File: src/pds/provenance/provenance.py (hunk 1 above)
Comment (alexdunnjpl) [reply src/pds/provenance/provenance.py:104]: This line was my first time too - thanks Al!

**Pull request merged**